app/Libraries/OsuAuthorize.php

```diff
@@ -326,6 +326,27 @@ public function checkBeatmapDiscussionPostStore($user, $post)
         return 'ok';
     }
 
+    public function checkBeatmapsetLove($user, $beatmapset)
+    {
+        $this->ensureLoggedIn($user);
+
+        static $prefix = 'beatmap_discussion.nominate.';
+
+        if (!$beatmapset->canBeLovedBy($user)) {
+            return 'unauthorized';
+        }
+
+        if (!$beatmapset->isLoveable()) {
```

Comment:
I'd move the beatmapset state check out and then combine `Beatmapset::canBeLovedBy` with this method, otherwise admins will be able to bypass the state check...